src/DynamoCoreWpf/Views/PackageManager/Controls/NumericUpDownControl.xaml.cs

```diff
@@ -0,0 +1,169 @@
+                typeof(int),
+                typeof(NumericUpDownControl),
+                new FrameworkPropertyMetadata(0, FrameworkPropertyMetadataOptions.BindsTwoWayByDefault, new PropertyChangedCallback(OnValuePropertyChanged)));
+
+        // Setting the input TextBox 
+        private static void OnValuePropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var numericUpDownControl = d as NumericUpDownControl;
+            numericUpDownControl.inputField.Text = e.NewValue.ToString();
+            if (numericUpDownControl.watermarkLabel.Visibility == Visibility.Visible)
+            {
+                numericUpDownControl.watermarkLabel.Visibility = Visibility.Collapsed;
+            }
+        } 
+        #endregion
+
+        public NumericUpDownControl()
+        {
+            InitializeComponent();
+
+            _numMatch = new Regex(@"^\d+$");
+        }
+
+        #region UI utility functions
+        private void spinnerUp_Click(object sender, RoutedEventArgs e)
+        {
+            if (string.IsNullOrEmpty(inputField.Text))
+            {
+                if (Int32.TryParse(watermarkLabel.Content as string, out int watermarkValue))
+                {
+                    Value = ++watermarkValue;
+                    return;
+                }
+                else
+                {
+                    inputField.Text = "0"; // increment from 0 if Watermark is anyhting else but integer
+                }
+            }
+            if (Int32.TryParse(inputField.Text, out int value))
+            {
+                Value = ++value;
+            };
+        }
+
+        private void spinnerDown_Click(object sender, RoutedEventArgs e)
+        {
+            if (string.IsNullOrEmpty(inputField.Text))
+            {
+                if (Int32.TryParse(watermarkLabel.Content as string, out int watermarkValue))
+                {
+                    Value = --watermarkValue;
+                    return;
+                }
+                else
+                {
+                    inputField.Text = "0"; // decrement from 0 if Watermark is anyhting else but integer
+                }
+            }
+            if (Int32.TryParse(inputField.Text, out int value))
+            {
+                Value = --value;
+            };
+        }
+
+        // validate data before changing the text (only allow numbers)
+        private void inputField_PreviewTextInput(object sender, TextCompositionEventArgs e)
+        {
+            var tb = (TextBox)sender;
+            var text = tb.Text.Insert(tb.CaretIndex, e.Text);
+
+            e.Handled = !_numMatch.IsMatch(text);
+        }
+
+        // Handles the direct input of text (via keyboard)
+        private void inputField_TextChanged(object sender, TextChangedEventArgs e)
+        {
+            if (Int32.TryParse(inputField.Text, out int value))
+            {
+                Value = value;
+            };
```

Comment:
I guess this means for non-number input to the version number fields. it does nothing, but don't we have a status display in the publish dialog to display some messages? Maybe it's worth considering displaying some help text there. @hwahlstrom

Reply:
I added an error message to Figma saying "Please enter a number" if they attempt to enter a non-number in the version fields.

Reply:
@hwahlstrom Thank you!

Reply:
@QilongTang to raise an exception if the user tries to enter non-numeric value?

Reply:
The control will not allow anything but positive whole numbers without explicit warning to the user at the moment. We can change that if it's necessary!

Reply:
@dnenov Thanks!